ob1k-core/src/main/java/com/outbrain/ob1k/client/targets/TargetProvider.java

```diff
@@ -1,11 +1,35 @@
 package com.outbrain.ob1k.client.targets;
 
+import java.util.List;
+
 /**
  * An API for retrieving targets for the client invocations.
  * An example target may be a static list of hosts + round robin, or a discovery based provider.
+ *
  * @author eran 6/21/15.
  */
 public interface TargetProvider {
-    String getTargetLogicalName();
-    String provideTarget();
+  /**
+   * Returns logical name of the target provider (e.g. client name).
+   *
+   * @return target logical name
+   */
+  String getTargetLogicalName();
+
+  /**
+   * Returns remote target for invocation.
+   *
+   * @return target host
+   */
+  String provideTarget();
+
+  /**
+   * Returns collection of remote targets for invocation for size specified.
+   * May return collection with duplicate targets, and the collection size will always
+   * be as requested
+   *
+   * @param targetsNum number of targets to return
+   * @return targets hosts
+   */
+  List<String> provideTargets(int targetsNum);
```

Comment:
@lifey not at all, thought it'll be nicer API
@haski the choose here of List is because duplicates are allowed, and order is important.
the first implementation was with Set of String, but after discussion with @eranharel, it's a possible case that a target provider would like to return two same hosts at sequence, for 'weight' case - if we want specific node to get more load than other ones, duplicating them solves this issue (for example, see ConsulBasedTargetProvider implementation).